File: spec/javascripts/hand_spec.js
describe('Hand', function(){

  describe('#valid', function(){
    describe('with 5 valid cards', function(){

      beforeEach(function(){
        var validCards = [
          new Card('heart', '10'),
          new Card('heart', 'J'),
          new Card('heart', 'Q'),
          new Card('heart', 'K'),
          new Card('heart', 'A')
        ];
        this.hand = new Hand(validCards);
      });

      it('returns true', function(){
        expect(this.hand.valid()).toBe(true);
      });
    });

    describe('with 4 cards', function(){

      beforeEach(function(){
        var invalidCards = [
          new Card('heart', '10'),
          new Card('heart', 'J'),
          new Card('heart', 'Q'),
          new Card('heart', 'K')
        ];
        this.hand = new Hand(invalidCards);
      });

      it('returns false', function(){
        expect(this.hand.valid()).toBe(false);
      });
    });
  });
});

File: Poker/hand_spec.rb
require 'rspec'
require_relative './hand'

describe Poker::Hand do
  subject(:hand) { Poker::Hand.new("AH 2C 3D 5S 6D") }
  let(:cards) { hand.cards }

  it "has five cards" do
    expect(cards.length).to equal 5
    cards.each do |card|
      expect(card).to be_an_instance_of Poker::Card
    end
  end
end
